workflow.Rmd

```diff
@@ -0,0 +1,473 @@
+
+
+# Running a project
+
+
+## Offline 
+This steps through the functions required to run a project using the original scripts. The aim of this script is to define project paths and set parameters used in the following scripts. 
+
+### Set up parameters and paths
+```{r 1_portfolio_check_initialisation, echo = FALSE}
+
+devtools::load_all()
+use_r_packages()
+
+## Project Initialisation
+
+working_location <- paste0(getwd(), "/")
+
+source("0_portfolio_test.R")
+source("0_graphing_functions.R")
+source("0_reporting_functions.R")
+source("0_portfolio_input_check_functions.R")
+source("0_global_functions.R")
+source("0_sda_approach.R")
+
+project_name <- "TestProject"
+twodii_internal <- FALSE
+# TRUE or FALSE: TRUE means that the code is running on a 2dii laptop with dropbox connection
+
+#####################################################################
+### ONLY FOR EXTERNAL PROJECTS (twodii_internal <- FALSE):
+# Variables must exist for internal projects
+project_location_ext <- "C:/Users/clare/Desktop/ExternalTest"
+data_location_ext <- "C:/Users/clare/Desktop/PACTA/pacta-data/2019Q4"
+#####################################################################
+
+create_project_folder(project_name, twodii_internal, project_location_ext)
+
+set_project_paths(project_name, twodii_internal, project_location_ext)
+
+copy_files(project_name)
+
+options(r2dii_config = fs::path(par_file_path, "AnalysisParameters.yml"))
+
+
+# parameters
+project_code <- "GENERAL"
+
+set_global_parameters(fs::path(par_file_path, "AnalysisParameters.yml"))
+
+set_project_parameters(file.path(working_location, "parameter_files",paste0("ProjectParameters_", project_code, ".yml")))
+
+analysis_inputs_path <- set_analysis_inputs_path(twodii_internal, data_location_ext, dataprep_timestamp)
+
+```
+
+TO DO: 
+
+- Improve the sourcing of files. Setting the internal flag to false to allow for defining this is one solution here. 
+
+- Synchronise the parameter files with the online parameter files
+
+
+
+### Portfolio input analysis
+
+Aims:
+
+- Reads in necessary data files and cleans them 
+
+To Run: 
+
+- You need a portfolio to be placed in the folder ```{r} fs::path(project_location_ext, project_name, "20_Raw_Inputs")```. There are sample files found in this repository in the folder "sample_files"
+
+Note:
+
+- Fund file optional. This adds additional holdings to the portfolio (adds in the details of the companies owned within a fund). If the file for the current time stamp is there this should be read in. Often this file is project specific as the file size can be very large. 
+
+- this is currently reading in the "cleaned" files found in the pacta-data repo. This workflow needs to be cleaned up so that these "cleaned" files can be created for any project.  
```

Comment:
Here it would be nice to include a code chunk showing that from the working directory we must be able to find the external repository pacta-data, and presumably some specific file in there.